ale/drivers/mex_drivers.py

```diff
@@ -322,77 +322,162 @@ def detector_center_sample(self):
     def line_scan_rate(self):
         """
         Returns a 2D array of line scan rates.
+    
+        For HRSC, the ephemeris times and exposure durations are 
+        stored in the image data. 
 
-        For HRSC, this data is actually imbedded in the binary data of the image
-        itself. Each line is stored in what is referred to as a "record" within
-        the image. The label will have the size of each record, the number of
-        records, and the number of records in the label, so the beginning of
-        binary data can be calculated.
+        Returns
+        -------
+        : list
+          Line scan rates
+        """
+        if not hasattr(self, '_line_scan_rate'):
+            times = [time - self.center_ephemeris_time for time in self.binary_ephemeris_times]
+            self._line_scan_rate = (self.binary_lines, 
+                                    times, 
+                                    self.binary_exposure_durations)
```

Comment:
Is there a reason why this is stateful?

Reply:
It was stateful before I refactored the code, I was not sure if I should remove it or not.

Reply:
The default should be non-stateful. Properties should only be stateful if they are time consuming to compute, have an external dependency such as a file, or have can be set.